### ISSUE TITLE
Enable custom splits in `DataSplitter`

### ISSUE THREAD

**Issue by martinkim0** [scverse/scvi-tools#2547]
See #1063. CC @canergen 

Two ways we could implement this:

`train_indices`, `validation_indices`, and `test_indices`
- Need to validate that indices are within range and check for overlaps
- Easier to adjust splits by changing indices
- Requires manual tracking of indices

`set_split_obs_key` that specifies an obs column in the data that should contain the values `"train"`, `"validation"`, and `"test"`
- Need to validate that this column only contains the correct values
- Potentially expensive for disk-backed datasets as we would need to load in this column for all observations prior to training (if we want to perform validation of the values). If a user wants to change the splits, they'll also have to write to disk
- Ties in splitting logic in with data so that this information is carried along with the dataset downstream
- One argument vs. 3

This feature will be important for integrating scib-metrics with the training loop as we want good representation of batches and cell types in the validation set.

**Comment by ori-kron-wis** [comment]
We'll go for the first option, as the user will just have to support his list of np.arrays of [train_idx,valid_idx,test_idx] by that order. He can leave any of them empty as well (as long as it makes sense).
I also added a flag for using external indexes. This will make sure the user wants to use those indexes, and it will bypass any logic we have today concerning splitting.
